source/core/module_coreHelpers.js

```diff
@@ -303,37 +303,40 @@
             return currentFPS;
         };
 
-        var writeNewError = function (newErrorStatus, newErrorCode, newErrorSource, existingErrorStatusPointer, existingErrorCodePointer, exisitingErrorSourcePointer) {
-            Module.eggShell.dataWriteBoolean(existingErrorStatusPointer, newErrorStatus);
-            Module.eggShell.dataWriteInt32(existingErrorCodePointer, newErrorCode);
-            Module.eggShell.dataWriteString(exisitingErrorSourcePointer, newErrorSource);
+        var writeNewError = function (errorValueRef, newError) {
+            Module.eggShell.writeJSON(errorValueRef, JSON.stringify(newError));
         };
 
-        Module.coreHelpers.mergeErrors = function (newErrorStatus, newErrorCode, newErrorSource, existingErrorStatusPointer, existingErrorCodePointer, exisitingErrorSourcePointer) {
+        Module.coreHelpers.mergeErrors = function (errorValueRef, newError) {
             // Follows behavior of merge errors function: https://zone.ni.com/reference/en-XX/help/371361N-01/glang/merge_errors_function/
+            if (errorValueRef === undefined) {
+                return;
+            }
 
-            var existingErrorStatus = Module.eggShell.dataReadBoolean(existingErrorStatusPointer);
+            var errorStatusValueRef = Module.eggShell.findSubValueRef(errorValueRef, 'status');
+            var existingErrorStatus = Module.eggShell.readDouble(errorStatusValueRef) !== 0;
```

Comment:
I think the variable existingErrorStatus is redundant. isExistingError  can be directly used.

Reply:
I'll remove it.